lib/kubeclient/exec_credentials.rb

```diff
@@ -34,6 +34,31 @@ def validate_opts(opts)
         raise KeyError, 'exec command is required' unless opts['command']
       end
 
+      def validate_client_credentials_status(status)
+        has_client_cert_data = status.key?('clientCertificateData')
+        has_client_key_data = status.key?('clientKeyData')
+
+        if has_client_cert_data && !has_client_key_data
+          raise 'exec plugin didn\'t return client key data'
+        end
+
+        if !has_client_cert_data && has_client_key_data
+          raise 'exec plugin didn\'t return client certificate data'
+        end
+
+        has_client_cert_data && has_client_key_data
+      end
+
+      def validate_credentials_status(status)
+        raise 'exec plugin didn\'t return a status field' if status.nil?
+
+        has_client_credentials = validate_client_credentials_status(status)
+        return if has_client_credentials
+
+        has_token = status.key?('token')
+        raise 'exec plugin didn\'t return a token' unless has_token
```

Comment:
At this point, I think you can say it returned neither token nor a client certificate.

Reply:
Done. Also added check to ensure a token XOR client creds are returned.